packages/puppeteer-core/src/bidi/Connection.ts

```diff
@@ -236,6 +240,15 @@ export class BidiConnection extends EventEmitter<BidiEvents> {
           return;
       }
     }
+    // Even if the response in not in BiDi protocol format but `id` is provided, reject
+    // the callback.
+    if ('id' in object) {
```

Comment:
Can you give more info here why this is needed? 
From my understanding we should always `type` to be populated from the spec perspective.

Reply:
If the WS endpoint supports CDP but not WS, there will be no `type` in the response. However, we still want the command promise to be rejected, so that we understand that the protocol is not supported.